stm32f0_hal/examples/uart.rs

```diff
@@ -1,23 +1,46 @@
 #![no_std]
 
+extern crate cortex_m;
 extern crate stm32f0_hal as hal;
+#[macro_use(interrupt)]
+extern crate stm32f0x2;
 
 use hal::{rcc, uart};
 
-fn main() {
-    rcc::init(); // Full Speed 48Mhz
+static mut UART1: Option<uart::Uart> = None;
+static mut DATA_UART1: u8 = 0;
 
-    uart::setup(
-        6000000,
+fn main() {
+    let uart3 = uart::Uart::setup(
+        uart::Uarts::Uart3,
+        57600,
         uart::NBits::_8bits,
         uart::StopBits::_1b,
         uart::Parity::None,
-        |byte| uart::send(byte),
```

Comment:
We could pass an option function argument more here to specify interrupt function.

Reply:
Also tracked in #25